New notebook example to use transformer for optimization

J-co opened the pull request:
One example file and a utility file that includes the interpolation scheme used to speed up propagation matrix calculation.

jhbdiehl commented:
That is a very nice addition! Thanks!

A few remarks on my watch:
- I think the two filenames should match more closely to indicate at first sight that they belong together. Sth along the lines "Transformer, Optimizing Disk Pos" and "Transformer_Optimizing_Disk_Pos_helper_funcs" (maybe a bit long?)?
- As always the notebook could use a little bit more explanation what is going on. (generally, e.g. what is the great new idea of this notebook, how does this help optimizing the bf much faster, and specifically, e.g. that you are optimizing only for 8 discreet frequencies, why is the p value you use -20, compare Stefans Notebooks)
- Your x_0 look suspiciously specific. Effectively they are flat distributed between -100e-6 and +100e-6, right?
- QN is failing for me unfortunately for f_tol = 1e-7. It also sometimes has strange behaviour claiming false successes (at < 100 iterations) when re-running. Did you encounter this as well?
- surface_roughness function unused?

J-co commented:
Thanks for the review.
I renamed the files to better indicate that they belong together. I also added an introduction to the notebook and some more comments.

What do you mean by suspiciously specific? I just use some random variation on the start spacing in case we chose a bad starting point.

The gradient seems to be somewhat unstable close to the minimum (thanks to the min() in the cost function i think). I changed f_tol and added a comment. Maybe this can be ironed out in the future...

I added an option to include surface roughness so the function is unused anymore ;)


I dont think I'm allowed to create branches (?)...

cheers,
Jacob

jhbdiehl commented:
> What do you mean by suspiciously specific? I just use some random variation on the start spacing in case we chose a bad starting point.

I just didn't understand at first why they looked the way they did. It's all right, especially since you clarified.

> I added an option to include surface roughness so the function is unused anymore ;)

Yay! After all that's more or less the point of the whole ordeal...

> I dont think I'm allowed to create branches (?)...

You are not? I should change that! Also I'm surprised that I am allowed to push to your fork.

I changed two minor things:
- Filename was not correctly updated in include statement at the beginning of the notebook
- I reduced ftol back to 1e-6, because 1e-4 seemed too relaxed to me. Final boosts were all over the place. Should be alright.

I'll merge now.
